texastribune/donations
# Notify via a Salesforce Task of Circle charge failures

#### What's this PR do?

It notifies them immediately when a Circle member's charge fails. 

#### Why are we doing this? How does it help us?

Terry wants to know right away so she can contact the member.

#### How should this be manually tested?

It's a little work. 
- add this to your `env-docker` file:
`CIRCLE_FAILURE_RECIPIENT=givalike@texastribune.org.stripeblas`
- start the container
- create a file, say `tmp.py`
- put this code in it:

```
from npsp import RDO, Contact
from charges import charge

contact = Contact.get(email="abc123@foobar.com")

rdo = RDO(contact=contact)

rdo.stripe_customer = "cus_E3k0SBDJim7sS1"
rdo.amount = 1
rdo.installments = 3
rdo.installment_period = "yearly"
rdo.type = "Giving Circle"
rdo.save()

opportunity = rdo.opportunities()[0]

charge(opportunity)

print(opportunity.stage_name)
print(opportunity.closed_lost_reason)
```
- Run that: `python tmp.py`
- Ask me to verify that a task got created. 

#### How should this change be communicated to end users?

Let Terry and Jenny know it's complete.

#### Are there any smells or added technical debt to note?

No.

#### What are the relevant tickets?

https://3.basecamp.com/3098728/buckets/736178/todos/1376643126

#### Have you done the following, if applicable:
***(optional: add explanation between parentheses)***

* [ ] Tested manually on mobile? *(N/A)*
* [ ] Checked for performance implications? *(negligible)*
* [ ] Updated the documentation/wiki? *(will on merge/deploy)*
